### PR TITLE
Update TensorFlow Micro lib w.r.t source code dated 20th Feb 2023

### PR DESCRIPTION
**Pull request by vibhor-m**
external/tfmicro: Update TensorFlow Micro lib dated 20thFeb2023 from OpenSource

build/configs/rtl8721csm/ai_tflite/Make.defs: Modify build flags
external/Kconfig: Remove duplicate entry
external/tfmicro: Update TensorFlow Micro v2.1.0 to TensorFlow Micro code dated 20th Feb 2023
apps/examples/hello_tfmicro: Changes to sample app for new TFLM
framework/src/lw_aifw/AIModel.cpp: Changes to AI Framework for new TFLM

external/include/libcxx: Modify complex and math.h to resolve compilation issue in Tensor Flow Micro
Files modified:
external/include/libcxx/complex
external/include/libcxx/math.h

### CODE REVIEW

File: apps/examples/hello_tfmicro/main.cc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2023 Samsung Electronics All Rights Reserved.
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
  *
```
Comment (Tirtha-Samsung) [inline apps/examples/hello_tfmicro/main.cc:4]: Why do we need to change the year to 2020?

File: apps/examples/hello_tfmicro/sine_model_data.h [hunk 2]
```diff
@@ -31,15 +13,15 @@ See the License for the specific language governing permissions and
...
-/* This is a standard TensorFlow Lite model file that has been converted into a
- C data array, so it can be easily compiled into a binary for devices that
- don't have a file system. It was created using the command:
- xxd -i sine_model.tflite > sine_model_data.cc
-*/
+// This is a standard TensorFlow Lite model file that has been converted into a
+// C data array, so it can be easily compiled into a binary for devices that
+// don't have a file system. It was created using the command:
+// xxd -i sine_model.tflite > sine_model_data.cc
+
```
Comment (Tirtha-Samsung) [inline apps/examples/hello_tfmicro/sine_model_data.h:20]: is multiple lines comment not better than this single line comment?

**Review by Taejun-Kwon — APPROVED** [review]

File: apps/examples/hello_tfmicro/Kconfig_ENTRY [hunk 3]
```diff
@@ -1,3 +1,3 @@
-config ENTRY_HELLO_TFMICRO
+config ENTRY_TFMICRO
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/Kconfig_ENTRY:1]: This looks wrong. Previous looks right.

File: apps/examples/hello_tfmicro/Make.defs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ###########################################################################
 #
-# Copyright 2023 Samsung Electronics All Rights Reserved.
+# Copyright 2020 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/Make.defs:3]: As @Tirtha-Samsung said, this should be 2023.

File: apps/examples/hello_tfmicro/main.cc [hunk 5]
```diff
@@ -33,125 +33,136 @@ limitations under the License.
...
-	/* Obtain pointers to the model's input and output tensors. */
-	input = interpreter->input(0);
-	output = interpreter->output(0);
-
-	/* Keep track of how many inferences we have performed. */
-	inference_count = 0;
+}  // namespace
+
+// The name of this function is important for Arduino compatibility.
+void setup() {
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/main.cc:66]: You should follow TizenRT coding rules in the apps folder.

File: build/configs/rtl8721csm/Make.defs [hunk 6]
```diff
@@ -171,6 +171,14 @@ ifeq ($(CONFIG_HAVE_CXX),y)
   CXXFLAGS += -pedantic -D_DEBUG -D_LIBCPP_BUILD_STATIC -ffunction-sections -fdata-sections -D__GLIBCXX__
 endif
 
+ifeq ($(CONFIG_EXTERNAL_TFMICRO),y)
+CXXFLAGS += -DTF_LITE_STATIC_MEMORY -DNDEBUG -g
+CFLAGS   += -DTF_LITE_STATIC_MEMORY -DNDEBUG -g
+CXXFLAGS += -I$(TOPDIR)/../external/tfmicro
+CXXFLAGS += -I$(TOPDIR)/../external/tfmicro/third_party
+CXXFLAGS += -I$(TOPDIR)/../external/tfmicro/third_party/gemmlowp
+endif
```
Comment (sunghan-chang) [inline build/configs/rtl8721csm/Make.defs:180]: Are they necessary here? not in the folder which uses TFMICRO like aifw?

File: external/include/libcxx/math.h [hunk 7]
```diff
@@ -1034,6 +1034,7 @@ copysign(_A1 __lcpp_x, _A2 __lcpp_y) _NOEXCEPT
     return copysign((__result_type)__lcpp_x, (__result_type)__lcpp_y);
 }
 
+#if 0
```
Comment (sunghan-chang) [inline external/include/libcxx/math.h:1037]: Why do you comment below? Please leave the reason with comment.

File: external/tfmicro/LICENSE [hunk 8]
```diff
@@ -187,7 +187,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2014 Google Inc.
+   Copyright [yyyy] [name of copyright owner]
```
Comment (sunghan-chang) [inline external/tfmicro/LICENSE:190]: Is this right?

File: external/tfmicro/Make.defs [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ###########################################################################
 #
-# Copyright 2023 Samsung Electronics All Rights Reserved.
+# Copyright 2022 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline external/tfmicro/Make.defs:3]: same as previous comment. You should check all of this kind of change.

**Review by sunghan-chang — COMMENTED** [review]
You modified many modules in a single commit. Please split them when you modify something like this next time.

File: build/configs/rtl8721csm/Make.defs (hunk 6 above)
Comment (vibhor-m) [reply build/configs/rtl8721csm/Make.defs:180]: these are included here, as it is required in build of framework/src/AIFW & apps/examples/aifw_test also.

File: apps/examples/hello_tfmicro/main.cc (hunk 5 above)
Comment (vibhor-m) [reply apps/examples/hello_tfmicro/main.cc:66]: This is sample application of TFMicro, I executed tools/cxxformatter.sh to make code guideline in sync with TizenRT

File: external/include/libcxx/math.h (hunk 7 above)
Comment (vibhor-m) [reply external/include/libcxx/math.h:1037]: I will add reason in a separate commit, along with other suggestion fix.

**Pull request merged**